Updates for Terraform policy docs (#2864)
hashicorp/web-unified-docs: content/terraform-policy/v0.1.x (beta)/docs/policy/reference/test/index.mdx

Each policy test file can include:

- An optional [`policytest` block](/terraform/policy/reference/test/policytest) to configure which policies Terraform policy will test.
- One or more mock blocks to define test cases by mocking the Terraform construct to test the policy against:
  - [`provider` blocks](/terraform/policy/reference/test/provider) to mock providers.
  - [`resource` blocks](/terraform/policy/reference/test/resource) to mock resources.
  - [`data` blocks](/terraform/policy/reference/test/data) to mock data sources.
  - [`module` blocks](/terraform/policy/reference/test/module) to mock modules.
- An optional [`inputs` block](/terraform/policy/reference/test/inputs) to set policy input values for all test cases.
- An optional [`locals` block](/terraform/policy/reference/test/locals) to define local values.

## Test block attributes

The following table lists the optional arguments that configuration blocks in test files support.

| Name             | Description | `provider` | `resource` | `data` | `module` |

> |------------------|----------|----------|------|--------|

Reviewer: The separator row has 5 cells but the header has 6 columns, so this table won't render.

```suggestion
|------------------|-------------|------------|------------|--------|----------|
```

  - [`resource {}` blocks](/terraform/policy/reference/test/resource) to mock resources.
  - [`data {}` blocks](/terraform/policy/reference/test/data) to mock data sources.
  - [`module {}` blocks](/terraform/policy/reference/test/module) to mock modules.

> - An optional [`inputs {}` block](/terraform/policy/reference/test/input) to set policy input values for all test cases.

Reviewer 2: ```suggestion
- An optional [`inputs {}` block](/terraform/policy/reference/test/inputs) to set policy input values for all test cases.
```

> - An optional [`inputs {}` block](/terraform/policy/reference/test/input) to set policy input values for all test cases.

Reviewer 3: Should the link be `/test/inputs`?

- An optional [`locals {}` block](/terraform/policy/reference/test/locals) to define local values.

## Test block attributes

> The following table lists the optional arguments supported by test blocks.

Reviewer 2: ```suggestion
The following table lists the optional arguments that configuration blocks in test files support.
```
Right? There isn't a `test` block, but there is a `policytest.hcl` file and these refer to the blocks in that file, correct? We should update the heading, too.

> | Name             | Description | Provider | Resource | Data | Module |

Reviewer 2: ```suggestion
| Name             | Description | `provider` | `resource` | `data` | `module` |
```
Your call, but I think formatting as the block keyword would help with the clarity.

## Testing operation-based policies

When a resource policy uses the `operations` argument to select Terraform lifecycle operations, your test cases must indicate which action each mock resource represents.

> The testing framework provides two optional attributes within mock resource blocks for this purpose: `meta.operation` and `prior_attrs`.

Reviewer 2: ```suggestion
The testing framework provides optional attributes within mock resource blocks for this purpose. The `meta.operation` declares which planned action a mock resource represents. The `prior_attrs` operation populates the mock data for the `prior_attrs` variables used by the policy evaluation context.
```

> ### The `meta.operation` attribute

Reviewer 2: ```suggestion
### `meta.operation` attribute
```
Optional. If you're linking to this section, then removing "The" from the heading probably isn't worth it.

> The testing framework provides optional attributes within mock resource blocks for this purpose. The `meta.operation` declares which planned action a mock resource represents. The `prior_attrs` operation populates the mock data for the `prior_attrs` variables used by the policy evaluation context.

Reviewer: "The `prior_attrs` operation populates..." -- should be "The `prior_attrs` attribute populates...".
